api/src/main/java/org/apache/iceberg/IncrementalScan.java

```diff
@@ -55,4 +80,23 @@
    * @return this for method chaining
    */
   ThisT toSnapshot(long toSnapshotId);
+
+  /**
+   * Instructs this scan to look for changes up to a particular snapshot ref (inclusive).
+   *
+   * <p>If the end snapshot is not configured, it is defaulted to the current table snapshot
+   * (inclusive).
+   *
+   * @param fromSnapshotRef the end snapshot Ref (inclusive)
+   * @return this for method chaining
+   */
+  ThisT toSnapshot(String fromSnapshotRef);
```

Comment:
I think we should just call this `ref` the method name already indicates it's "to"